NSS-Day-Cohort-42/rare-media-moguls
# Rp add posts

#### Changes Made
1. created PostProvider.js with getPosts and addPost functions
2.created PostForm.js with form to create a new post
3. added PostForm to application views and navbar
​
#### Steps to Review
1. Checkout this branch locally and run the application.
2. Check that posts are created by opening network tab in dev tools
3. redirecting users up clicking save post button will be included in later pr
```
git fetch --all
git checkout rp-addPost
```
2. Run the application.
```
npm start
```
3. Open a new Terminal tab (⌘T) and navigate to the server directory.
```
..
server
```
4. Initialize virtual environment, and run the server.
```
pipenv shell
watchgod request_handler.main
```
5. Test app functionality. 
> When user clicks on `New Post`, they should be redirected to a the PostForm page
> When user enters information into the form fields and clicks `Save`, the post database should be updated with new post object and user should be redirected to the PostDetails page to view the post they just created.
6. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.
